diff --git a/crates/prime-compositor/src/input.rs b/crates/prime-compositor/src/input.rs
--- a/crates/prime-compositor/src/input.rs
+++ b/crates/prime-compositor/src/input.rs
@@ -4,7 +4,7 @@ use smithay::{
         Axis, AxisSource, Event, InputBackend, InputEvent, KeyboardKeyEvent, PointerAxisEvent,
         PointerButtonEvent, PointerMotionEvent,
     },
-    desktop::{layer_map_for_output, WindowSurfaceType},
+    desktop::{find_popup_root_surface, layer_map_for_output, WindowSurfaceType},
     input::{
         keyboard::FilterResult,
         pointer::{AxisFrame, ButtonEvent, MotionEvent},
@@ -221,5 +221,8 @@ fn keyboard_focus_under(runtime: &Runtime, location: Point<f64, Logical>) -> Option<WlSurface> {
     while let Some(parent) = get_parent(&root) {
         root = parent;
     }
+    if let Some(popup) = runtime.protocols.popups.find_popup(&root) {
+        return find_popup_root_surface(&popup).ok();
+    }
     Some(root)
 }
